### PR TITLE
MDS-1920 - Update Mine navigation

### PR DESCRIPTION
**Pull request by mmumby**
Opening to start the code review process. 
The majority of the changes are converting components into route components and updating the way props are passed in.
 
Files/changes ready for review
- `MineHeader.js`
- `MineSummary.js`
- `MineContactinfo.js`
- `MineApplicationInfo.js`
- `RefreshButton.js`
- `MineVariance`
- `MineIncidents`
- `MineTailingsInfo`
- `MinePermitInfo`
- `MineReportInfo`
- `MineComplianceInfo`
- `MineDashboard.js`
- Update all Links to that go to the Mine dashboard view
- Update styling and add active links css for the new menu navigation
**- functional tests have all been updated to work with the new navigation**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@ __pycache__
 *.pgCustom
 *.sw*
 python-backend/document_uploads/
+microservices/document_manager/backend/document_uploads/
```
Comment (NWCalvank) [inline .gitignore]: So, did this end up working?

File: frontend/src/components/Forms/reports/AddReportForm.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (NWCalvank) [inline frontend/src/components/Forms/reports/AddReportForm.js]: Explain?

File: frontend/src/components/common/RenderYear.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (NWCalvank) [inline frontend/src/components/common/RenderYear.js]: Also this?

File: frontend/src/components/mine/Compliance/MineComplianceInfo.js [hunk 4]
```diff
@@ -1,37 +1,138 @@
+/* eslint-disable */
```
Comment (NWCalvank) [inline frontend/src/components/mine/Compliance/MineComplianceInfo.js]: ?

File: frontend/src/components/mine/Reports/MineReportActions.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (NWCalvank) [inline frontend/src/components/mine/Reports/MineReportActions.js]: ?

File: frontend/src/components/mine/Reports/MineReportTable.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (NWCalvank) [inline frontend/src/components/mine/Reports/MineReportTable.js]: ?

File: frontend/src/components/mine/Compliance/MineComplianceInfo.js (hunk 4 above)
Comment (mmumby) [reply frontend/src/components/mine/Compliance/MineComplianceInfo.js]: will remove from this file!

File: .gitignore (hunk 1 above)
Comment (mmumby) [reply .gitignore]: yes it did!

**Pull request merged**